src/bsoncxx/array/value.hpp

```diff
@@ -26,14 +26,49 @@ namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 namespace array {
 
+///
+/// A read-only BSON array that owns its underlying buffer. When a array::value goes
+/// out of scope, the underlying buffer is freed. Generally this class should be used
+/// sparingly; array::view should be used instead wherever possible.
+///
 class BSONCXX_API value {
 
    public:
     using deleter_type = void(*)(void*);
     using unique_ptr_type = std::unique_ptr<uint8_t, deleter_type>;
 
+    ///
+    /// Constructs a value from a buffer.
+    /// This constructor transfers ownership of the buffer to the resulting
+    /// value. A user-provided deleter is used to destroy the buffer.
+    ///
+    /// @param data
+    ///   A pointer to a buffer containing a valid BSON array.
+    /// @param length
+    ///   The length of the document.
+    /// @param dtor
+    ///   A user provided deleter.
     value(std::uint8_t* data, std::size_t length, deleter_type dtor);
+
+    ///
+    /// Constructs a value from a std::unique_ptr to a buffer. The ownership
+    /// of the buffer is transferred to the resulting value.
+    ///
+    /// @param ptr
+    ///   A pointer to a buffer containing a valid BSON array.
+    /// @param length
+    ///   The length of the document.
+    ///
     value(unique_ptr_type ptr, std::size_t length);
+
+    ///
+    /// Constructs a value from a view of another array. The data referenced
```

Comment:
This is a nit but could be "from the view of an array"  instead of "another array" because they ultimately are represent the same data right?